src/py/flwr/common/auth_plugin/auth_plugin.py

```diff
@@ -0,0 +1,91 @@
+# Copyright 2024 Flower Labs GmbH. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Abstract classes for Flower User Auth Plugin."""
+
+from abc import ABC, abstractmethod
+from collections.abc import Sequence
+from pathlib import Path
+from typing import Any, Union
+
+import grpc
+
+from flwr.proto.exec_pb2 import (  # pylint: disable=E0611
+    GetAuthTokenRequest,
+    GetAuthTokenResponse,
+    LoginResponse,
+)
+from flwr.proto.exec_pb2_grpc import ExecStub
+
+
+class ExecAuthPlugin(ABC):
+    """Abstract Flower Auth Plugin class for ExecServicer."""
+
+    @abstractmethod
+    def __init__(self, config: dict[str, Any]):
+        """Abstract constructor (init)."""
+
+    @abstractmethod
+    def get_login_response(self) -> LoginResponse:
+        """Send relevant login details as a LoginResponse."""
```

Comment:
```suggestion
```suggestion
    def login(self) -> LoginResponse:
        """Send the login details and return the server's response."""
```
```

Reply:
This is only the login url that is being sent back to the user. The login process itself does not happen here.

Reply:
Oh, I see. Thanks for the clarification. Could we then:

```suggestion
    def get_login_response(self) -> LoginResponse:
        """Get the LoginResponse containing the login details."""
```